controllers/druid/handler.go

```diff
@@ -547,6 +553,130 @@ func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
+		msg := fmt.Sprintf("failed to get [StatefuleSet:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+		logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+		sendEvent(sdk, drd, v1.EventTypeNormal, "GET_FAIL", err.Error())
+		return
+	}
+
+	pod := makePodEmptyObj()
+	if err := sdk.Get(context.TODO(), *namespacedName(nodeSpecUniqueStr, drd.Namespace), pod); err != nil {
+		msg := fmt.Sprintf("failed to get [pod:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+		logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+		sendEvent(sdk, drd, v1.EventTypeNormal, "GET_FAIL", err.Error())
+		return
+	}
+
+	// ignoring error, function shall log and sendEvent for it
+	pvc, _ := getPVCList(sdk, drd)
+
+	desVolumeClaimTemplateSize, currVolumeClaimTemplateSize, pvcSize := getVolumeClaimTemplateSizes(sts, nodeSpec, pvc)
+
+	for i := range desVolumeClaimTemplateSize {
+
+		// Validate Request, shrinking of pvc not supported
+		// desired size cant be less than current size
+		// in that case re-create sts/pvc which is a user execute manual step
+		if desVolumeClaimTemplateSize[i].String() < currVolumeClaimTemplateSize[i].String() {
+			e := fmt.Errorf("Request for Shrinking of sts pvc size [sts:%s] in [namespace:%s] is not Supported", sts.Name, sts.Namespace)
+			logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+			sendEvent(sdk, drd, v1.EventTypeWarning, "STS_STS_PVC_RESIZE_FAIL", e.Error())
+			return
+		}
+
+		// In case size dont match and dessize > currsize, delete the sts using casacde=false / propagation policy set to orphan
+		// The operator on next reconcile shall create the sts with latest changes
+		if desVolumeClaimTemplateSize[i].String() != currVolumeClaimTemplateSize[i].String() {
+			msg := fmt.Sprintf("Detected Change in VolumeClaimTemplate Sizes for Statefuleset [%s] in Namespace [%s], desVolumeClaimTemplateSize: [%s], currVolumeClaimTemplateSize: [%s]\n, deleteing STS [%s] with casacde=false]", sts.Name, sts.Namespace, desVolumeClaimTemplateSize[i].String(), currVolumeClaimTemplateSize[i].String(), sts.Name)
+			logger.Info(msg)
+			sendEvent(sdk, drd, v1.EventTypeNormal, "STS_SIZE_DETECTED", msg)
+
+			if err := sdk.Delete(context.TODO(), sts, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
+				e := fmt.Errorf("failed to Delete [StatefuleSet:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+				logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+				sendEvent(sdk, drd, v1.EventTypeWarning, "DELETE_FAIL", e.Error())
+			} else {
+				msg := fmt.Sprintf("[StatefuleSet:%s] successfully deleted with casacde=false", sts.Name)
+				logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+				sendEvent(sdk, drd, v1.EventTypeNormal, "DELETE_SUCCESS", msg)
+			}
+		}
+
+		// In case size dont match, patch the pvc with the desiredsize from druid CR
+		if desVolumeClaimTemplateSize[i].String() != pvcSize[i].String() {
```

Comment:
Possibilities of a panic here, the array count in desVolumeClaimTemplateSize can be greater than > pvc Size count, which can cause the for loop iteration to fail. On the other hand, array count of pvcSize can be greater than desVolumeClaimTemplateSize since one sts can have multiple replicas each having a pvc. So ideally iteration to take place separately for pvcSize and the index. 
There should be a check if len(pvcSize) <  len(desVolumeClaimTemplateSize) return